### PR TITLE
Fiducial models and tilt-series methods

### PR DESCRIPTION
**Pull request by fede-pe**
This PR includes

- New method in TiltSeriesBase object for .tlt angle file generation
- New method in TiltSerie object to apply a transformation matrix
- New object LandmarkModel
- New object SetOfLandmarkModel
- New method to create SetOfLandmarkModel in protocol_base.py

### CODE REVIEW

File: tomo/objects.py [hunk 1]
```diff
@@ -122,10 +125,44 @@ def getScannedPixelSize(self):
         mag = self._acquisition.getMagnification()
         return self._samplingRate.get() * 1e-4 * mag
 
+    def generateTltFile(self, tltFilePath, reverse=False):
```
Comment (pconesa) [inline tomo/objects.py:128]: @fede-pe could you add a description for the method. I know what is doing but cannot see why this file with angles is needed. Is it for a specific software, when this file is needed?

File: tomo/objects.py [hunk 2]
```diff
@@ -122,10 +125,44 @@ def getScannedPixelSize(self):
...
+                    transform = ti.getTransform().getMatrix()
+                    transformArray = np.array(transform)
+                    ih.applyTransform(inputFile=str(index + 1) + ':mrcs@' + inputFilePath,
+                                      outputFile=str(index + 1) + '@' + outputFilePath,
+                                      transformMatrix=transformArray,
+                                      shape=(ti.getXDim(), ti.getYDim()))
+                else:
+                    raise Exception('ERROR: Some tilt-image is missing from transform object associated.')
+        else:
+            path.createLink(self.getFirstItem().getLocation()[1], outputFilePath)
```
Comment (pconesa) [inline tomo/objects.py:164]: I think getFileName() reads better than getLocation()[1] and should work.

File: tomo/objects.py [hunk 3]
```diff
@@ -791,7 +828,63 @@ class SetOfClassesSubTomograms(data.SetOfClasses):
...
+        self._tsId = pwobj.String(tsId)
+
+    def setFileName(self, fileName):
+        self._fileName = pwobj.String(fileName)
+
+    def setModelName(self, modelName):
+        self._modelName = pwobj.String(modelName)
+
+    def addLandmark(self, xCoor, yCoor,  tiltIm, chainId, xResid, yResid):
+        import csv
```
Comment (pconesa) [inline tomo/objects.py:859]: I'd refactor this a bit:

1st: imports should be at the top unless there is a good reason not to (import endless loops)
2nd:  I'd try to have 1 single line of "writerow". and  DictWriter line:
````
mode = "a" if os.path.exists(self.getFileName()) else "w"

with open(self.getFileName(), mode) as f:

    if mode == "w"
        writer.writeheader()
    writer = csv.DictWriter(f, delimiter='\t', fieldnames=fieldNames)
    ....

````

File: tomo/objects.py (hunk 1 above)
Comment (fede-pe) [reply tomo/objects.py:128]: .tlt angle files are a usual input for different programs of different plugins (as IMOD or NovaCtf). Thus, this method generates a file with the angles of the tilt series from the Scipion object, that can be served to these programs. I have added the requested description.

File: tomo/objects.py (hunk 2 above)
Comment (fede-pe) [reply tomo/objects.py:164]: You right. Done.

Comment (pconesa) [reply tomo/objects.py:164]: Push?

File: tomo/objects.py (hunk 3 above)
Comment (fede-pe) [reply tomo/objects.py]: Done

File: tomo/objects.py (hunk 2 above)
Comment (fede-pe) [reply tomo/objects.py]: Now should be.

**Pull request merged**